src/derivative_utils.jl

```diff
@@ -1,3 +1,26 @@
+mutable struct SemiImplicitNLSolver{WType,JType,du1Type,ufType,jcType}
```

Comment:
I guess, `NLSemiImplicitCache` (or something along those lines) would be more consistent with DiffEqBase. It's not an alternative to `NLSolver` but just a stripped down version of `NLNewtonCache`, it seems?

Reply:
It is basically just a cache, just a container, but it needs to be saved by the name `nlsolver` in cache because it is used as a nlsolver in `derivative_utils.jl`. Its like making `derivative_utils.jl` think like this is an actual nlsolver so it fetches the required fields in the same way it would fetch from a nlsolver.

Reply:
I think we should make it a cache instead, `derivative_utils.jl` shouldn't have to work with the solvers at all after the changes in DiffEqBase. There's nothing contained anymore in `nlsolver` that could be of interest to `derivative_utils.jl` apart from the cache, I think?

Reply:
No, problem lies for threaded algorithms. We can have multiple nlsolvers so specifying nlsolver is needed.

Reply:
Hmm, I don't see why you.would.want to pass around the (Pseudo)NLsolver structs. To me it seems passing multiple caches should.be sufficient.